Merge wxAdvanced into wxCore

vadz opened the pull request:
AFAICS this is the minimum required. With these changes, I can build wxGTK, wxX11 (i.e. wxUniv build) and wxMSW as DLLs.

Please let me know if you see anything I forgot or if you disagree with the `dummy.cpp` hack (I'm on a fence about this one, on one hand it's ugly, but OTOH it doesn't seem worth introducing another dummy source file just for this, how much dumminess do we want to have).

vadz committed:
Move wxAdv library contents into wxCore

This basically removes the "adv" library, even though it's still
preserved for compatibility with user make/project files referring to
it.

It is done because the distinction between "adv" and "core" was never
really clear (e.g. why wxTreeCtrl was in core but wxTreeListCtrl in
adv?) and it prevented some core classes from using adv ones.

vadz committed:
Fix wrong DLL in the declaration of wxPendingDelete

This object is defined in "base", not "core".

Somehow this still worked before, but it was still wrong and stopped
working when taskbarcmn.cpp was moved to "core" itself from "adv".

vadz committed:
Force creating import library for "adv" DLL under MSW

No library was created as no symbols were exported by the DLL any more,
so add a dummy symbol just to force the import library creation -- this
is important to let the projects referencing it to continue to link,
even if it's now useless.

vadz committed:
Remove references to "adv" library from MSVS 201x projects

They are not generated by bakefile, so need to be updated manually.

vadz committed:
Completely remove "adv" library from CMake build system

This build system is new and so doesn't need to keep "adv" for
compatibility.

MaartenBent commented:
There is some mentioning if `adv` in
`misc\scripts\ctags.ignore`
`docs\doxygen\mainpages\screenshots.h`
`docs\doxygen\mainpages\const_cpp.h`
And a lot of `@library{wxadv}` in the interface headers.

vadz commented:
Good points, thanks @MaartenBent, will fix.

vadz committed:
Don't use a separate section for "adv" library controls

Put them together with the other "miscellaneous" controls now that the
"adv" library doesn't exist any more.

vadz committed:
Remove "adv" library from MSVS-specific header

Also stop documenting wxNO_ADV_LIB as it has no effect any more.

vadz committed:
Replace references to "adv" with "core" in classes documentation

All classes previously in "wxadv" are now in "wxcore".

vadz commented on build/cmake/samples/CMakeLists.txt:
```diff
@@ -8,22 +8,22 @@
 
 wx_add_sample(access accesstest.cpp DEPENDS wxUSE_ACCESSIBILITY)
-wx_add_sample(animate anitest.cpp anitest.h LIBRARIES adv DATA throbber.gif hourglass.ani DEPENDS wxUSE_ANIMATIONCTRL)
+wx_add_sample(animate anitest.cpp anitest.h DATA throbber.gif hourglass.ani DEPENDS wxUSE_ANIMATIONCTRL)
```
If I understand CMake syntax correctly (a big if, admittedly), `LIBRARIES` is a keyword and so is not needed (or maybe even harmful?) if there is nothing following it.

MaartenBent replied:
Not harmful, but other optional keywords are omitted as well when they have no values.
